packages/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx

```diff
@@ -29,5 +29,5 @@ export const GRID_DETAIL_PANEL_TOGGLE_COL_DEF: GridColDef = {
     return expandedRowIds.includes(rowId);
   },
   renderCell: (params) => <GridDetailPanelToggleCell {...params} />,
-  renderHeader: () => null,
+  renderHeader: (params) => <div aria-label={params.colDef.headerName} />,
```

Comment:
Before, the condition `headerComponent == null ? label : undefined` in `GridGenericColumnHeaderItem` was ensuring that the aria label is added to this column without actually rendering anything.
Now the column definition holds that logic in itself

Reply:
How about we do this internally instead?

```diff
diff --git a/packages/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx b/packages/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx
index 0fc2a55a78..ce3a18b882 100644
--- a/packages/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx
+++ b/packages/x-data-grid-pro/src/hooks/features/detailPanel/gridDetailPanelToggleColDef.tsx
@@ -29,5 +29,5 @@ export const GRID_DETAIL_PANEL_TOGGLE_COL_DEF: GridColDef = {
     return expandedRowIds.includes(rowId);
   },
   renderCell: (params) => <GridDetailPanelToggleCell {...params} />,
-  renderHeader: (params) => <div aria-label={params.colDef.headerName} />,
+  renderHeader: () => null,
 };
diff --git a/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx b/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx
index 412bb46be9..91ecd13dc5 100644
--- a/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx
+++ b/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx
@@ -120,7 +120,7 @@ const GridGenericColumnHeaderItem = React.forwardRef(function GridGenericColumnH
         <div className={classes.titleContainer} role="presentation">
           <div className={classes.titleContainerContent}>
             {headerComponent !== undefined ? (
-              headerComponent
+              headerComponent || <div aria-label={label} />
             ) : (
               <GridColumnHeaderTitle label={label} description={description} columnWidth={width} />
             )}

```

Reply:
I would rather keep it with the definition that actually needs it and keep the generic component clean from it.
Otherwise we have this (for me) weird algorithm where we say:
- If you don't provide a component a fallback will be rendered
- Else If your component is `null` another fallback will be rendered
- Else your component is rendered

Searching the code, I didn't find any other place where `null` is returned (there is another definition with `' '`, but that will not end up as this special condition)

Do you think that someone relies on the current logic and that the things will break?

Reply:
I was mostly considering compatibility here, but also a better default – no need to know this trick to make the column header accessible.
The empty column header is a rare case though.
I don't have a strong opinion on this one, I'm OK with your approach 👍🏻